src/main.js

```diff
@@ -20,39 +21,80 @@ const footerStatisticsElement = footerMainElement.querySelector('.footer__statis
+  const filmComponent = new CardFilmView(film);
+  const filmPopupComponent = new PopupFilmView(film);
+
+  const replaceCardFilmToPopup = () => {
+    filmListEl.replaceChild(filmPopupComponent.element, filmComponent.element);//Замена одного компонента корточки на попап
+  };
+
+  const replacePopupToCardFilm = () => {
+    filmListEl.replaceChild(filmComponent.element, filmPopupComponent.element);//замена через ReplaceChild компомент попап на карточки
+  };
+
+  const onEscKeyDown = (evt) => {
+    if(evt.key === 'Escape' || evt.key === 'Esc') {
+      evt.preventDefault();
+      replacePopupToCardFilm();
+      document.removeEventListener('keydown', onEscKeyDown);
+      document.body.classList.remove('hide-overflow');
+    }
+  };
+
+  filmComponent.element.querySelector('.film-card__link').addEventListener('click', () => {
+    replaceCardFilmToPopup();
+    document.body.classList.add('hide-overflow');
+    document.addEventListener('keydown', onEscKeyDown);
+  });
+
+  filmPopupComponent.element.querySelector('.film-details__close-btn').addEventListener('click', () => {
+    replacePopupToCardFilm();
+    document.body.classList.remove('hide-overflow');
+    document.removeEventListener('keydown', onEscKeyDown);
+  });
+
+  render (filmListEl, filmComponent.element, RenderPosition.BEFOREEND);
+};
+
+if(films.length === 0) {
+  render(filmsListContainerElement, new MessageFilmsListEmptyView().element, RenderPosition.BEFOREEND);
+}
+
 for (let i = 0; i < Math.min(films.length, FILM_COUNT_PER_STEP); i++) {
-  renderTemplate(filmsListContainerElement, createCardFilmTemplate(films[i]), RenderPosition.BEFOREEND); //карточки фильмов
+  renderFilms(filmsListContainerElement, films[i]); //карточки фильмов
 }
 
 const filmsListElement = filmsElement.querySelector('.films-list');
 if (films.length > FILM_COUNT_PER_STEP) {
   let renderFilmsCount = FILM_COUNT_PER_STEP;
-  renderTemplate(filmsListElement, createBtnShowMoreTemplate(), RenderPosition.BEFOREEND);
+  render(filmsListElement, new BtnShowMoreView().element, RenderPosition.BEFOREEND);
 
   const showMoreButton = filmsListElement.querySelector('.films-list__show-more');
 
   showMoreButton.addEventListener('click', (evt) => {
     evt.preventDefault();
     films
       .slice(renderFilmsCount, renderFilmsCount + FILM_COUNT_PER_STEP)
-      .forEach((film) => renderTemplate(filmsListContainerElement, createCardFilmTemplate(film), RenderPosition.BEFOREEND));
+      .forEach((film) => renderFilms(filmsListContainerElement,film));
     renderFilmsCount += FILM_COUNT_PER_STEP;
 
     if (renderFilmsCount >= films.length) {
       showMoreButton.remove();
     }
   });
 }
-renderTemplate(footerStatisticsElement, createStatistiscFilmsTemplates(), RenderPosition.BEFOREEND);
-renderTemplate(footerMainElement, createFilmDetailsPopupTemplates(films[0]), RenderPosition.BEFOREEND);
+
+render(footerStatisticsElement, new StatisticsView().element, RenderPosition.BEFOREEND);
```

Comment:
Создание компонентов лучше присваивать переменной, они могут нам пригодиться в будущем
```suggestion
const staticView = new StaticView();
render(footerStatisticsElement, staticView.element, RenderPosition.BEFOREEND);
```